File: Learning Android App/Internship/app/src/main/java/com/example/internship/MainActivity.java
package com.example.internship;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonArrayRequest;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;
import com.example.internship.databinding.ActivityMainBinding;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private ArrayList<RepoModel> repoModelArrayList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.findButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (binding.userName.getText().toString().trim().isEmpty()) {
                    Toast.makeText(MainActivity.this, "Enter the user name", Toast.LENGTH_SHORT).show();
                } else {
                    Intent intent = new Intent(MainActivity.this, RepoList.class);
                    intent.putExtra("userName", binding.userName.getText().toString().trim());
                    startActivity(intent);
                    Log.d("Main", binding.userName.getText().toString().trim());
                }
            }
        });

    }


}

File: Learning Android App/Kotlin_Java_Mix/app/src/main/java/com/example/kotlin_java_mix/MainActivity.kt
package com.example.kotlin_java_mix

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.kotlin_java_mix.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)


        binding.kotlin.setOnClickListener {
            binding.textView.text = "hello from kotlin"
        }
        binding.java.setOnClickListener {
            val javaHelper = Helper()
            javaHelper.updateFromJava(binding.textView)
        }

    }
}